pkg/platform/runtime/setup/setup.go

```diff
@@ -212,19 +247,24 @@ func (s *Setup) Update() error {
 	return nil
 }
 
-func (s *Setup) updateFromRecipe() ([]artifact.ArtifactID, error) {
-	var noArtifacts []artifact.ArtifactID // in case of errors
+func (s *Setup) fetchArtifacts(installFunc artifactInstaller) error {
```

Comment:
Nit: It seems unnecessary that this function has an opinion on what the callback is used for (talking about the naming of the arg and type).

Reply:
I could go with `callback FetchCallback`, but it's not as readable to me when used in-line. `callback(...)` vs. `installFunc(...)`. If you have a strong preference, I can do the former.